core/src/test/java/org/elasticsearch/client/transport/TransportClientIT.java

```diff
@@ -97,4 +106,47 @@ public void testThatTransportClientSettingCannotBeChanged() {
             assertThat(Client.CLIENT_TYPE_SETTING_S.get(settings), is("transport"));
         }
     }
+
+    /**
+     * test that when plugins are provided that want to register
+     * {@link NamedWriteable}, those are also made known to the
+     * {@link NamedWriteableRegistry} of the transport client
+     */
+    public void testPluginNamedWriteablesRegistered() {
+        Settings baseSettings = Settings.builder()
+                .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir())
+                .build();
+        try (TransportClient client = new MockTransportClient(baseSettings, pluginList(MockPlugin.class))) {
+            NamedWriteableRegistry registry = client.injector.getInstance(NamedWriteableRegistry.class);
```

Comment:
Actually if you make the NamedWriteableRegistry a member we could use it to fix the horrible, horrible hackery I added [here](https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java#L632-L645).